### PR TITLE
Created Success page

### PR DESCRIPTION
**Pull request by louiscoste**
image
image

### CODE REVIEW

**Comment by quinnbudan** [comment]
On the mockup, the "Successfully Created" text looks a little bit smaller than it is here. I'm not sure if we should make it smaller or not but just something I noticed

**Comment by louiscoste** [comment]
@quinnbudan I am using the D2L heading styles (in this case h2). I found using h3 was a little too small
This is what h3 looks like:
image

**Comment by TomMichielsD2L** [comment]
> @quinnbudan I am using the D2L heading styles (in this case h2). I found using h3 was a little too small
> This is what h3 looks like:
> image

I vote for h3

File: src/components/d2l-teacher-course-creation.js [hunk 1]
```diff
@@ -45,6 +48,10 @@ class TeacherCourseCreation extends BaseMixin(LitElement) {
 		if (event.detail && event.detail.page) {
 			this.currentPage = event.detail.page;
 		}
+
+		if (this.currentPage === PAGES.SUCCESS_PAGE) {
+			this.generatedOrgUnitId = '6609';
+		}
```
Comment (TomMichielsD2L) [inline src/components/d2l-teacher-course-creation.js:54]: Did you intend to commit this?

File: src/components/d2l-teacher-course-creation.js [hunk 2]
```diff
@@ -14,6 +14,9 @@ class TeacherCourseCreation extends BaseMixin(LitElement) {
 		return {
 			currentPage: {
 				type: String
+			},
+			generatedOrgUnitId: {
```
Comment (TomMichielsD2L) [inline src/components/d2l-teacher-course-creation.js:18]: What if we have an object like PageData that we populate and pass to each subcomponent. Then they can pull whatever they need out of it.

File: src/components/widget/d2l-teacher-course-creation-success.js [hunk 3]
```diff
@@ -1,23 +1,53 @@
...
+			.tcc-success-title-icon {
+				margin: 12px;
+				color: var(--d2l-color-feedback-success);
+				flex-shrink: 0;
+			}
+			.tcc-success-links-container {
+				display: flex;
+				flex-direction: column;
+				margin-bottom: 1.5rem;
+				border: 2px solid var(--d2l-color-gypsum);
```
Comment (TomMichielsD2L) [inline src/components/widget/d2l-teacher-course-creation-success.js:42]: maybe 1px?

File: src/components/widget/d2l-teacher-course-creation-success.js [hunk 4]
```diff
@@ -30,11 +60,50 @@ class TeacherCourseCreationSuccess extends BaseMixin(LitElement) {
...
+				<d2l-icon
+					class="tcc-success-title-icon"
+					icon="tier1:check-circle">
+				</d2l-icon>
+				<h1 class="d2l-heading-2">${this.localize('successTitleText')}</h1>
+			</div>
+
+			<div class="tcc-success-links-container">
+				<d2l-link
+					href=${'d2l/lms/classlist/classlist.d2l?ou='.concat(this.generatedOrgUnitId)}>
```
Comment (TomMichielsD2L) [inline src/components/widget/d2l-teacher-course-creation-success.js:83]: I would refactor the href construction to a function

File: src/components/widget/d2l-teacher-course-creation-success.js [hunk 5]
```diff
@@ -30,11 +60,50 @@ class TeacherCourseCreationSuccess extends BaseMixin(LitElement) {
...
+					${this.localize('successLinkEnrollStudents')}
+				</d2l-link>
+				<d2l-link
+					href=${this.generatedOrgUnitId}>
```
Comment (TomMichielsD2L) [inline src/components/widget/d2l-teacher-course-creation-success.js:87]: This needs a real link

File: src/components/widget/d2l-teacher-course-creation-success.js (hunk 3 above)
Comment (louiscoste) [reply src/components/widget/d2l-teacher-course-creation-success.js:42]: After zooming onto the mockup and comparing the 1px and 2px, I believe 2px is closer. Might need to have someone like Chris to make the final decision.

File: src/components/widget/d2l-teacher-course-creation-success.js (hunk 5 above)
Comment (louiscoste) [reply src/components/widget/d2l-teacher-course-creation-success.js:87]: It appends the orgUnitId to the base URL. which leads to the class homepage

File: locales/en.js [hunk 6]
```diff
@@ -5,6 +5,13 @@ export const val = {
 	'columnPrefix': 'Prefix',
 	'columnSuffix': 'Suffix',
 	'columnRole': 'Role',
+	'successDoneButtonDescription': 'Done creating course',
+	'successDoneButtonText': 'Done',
+	'successLinkEnrollStudents': 'Enroll students in course classlist',
+	'successLinkCourseHomepage': 'Go to course homepage',
+	'successLinkCreateCourse': 'Create another course',
+	'successResultText': 'Your new empty course has been created. You may enroll students into your course, or visit the main course home page or create another course.',
```
Comment (TomMichielsD2L) [inline locales/en.js:13]: This string contains the course name in the mockup. Should we add that?

File: src/components/widget/d2l-teacher-course-creation-success.js (hunk 4 above)
Comment (josephsaffron) [reply src/components/widget/d2l-teacher-course-creation-success.js]: and you'll need a leading /

File: src/components/widget/d2l-teacher-course-creation-success.js [hunk 7]
```diff
@@ -28,13 +61,64 @@ class TeacherCourseCreationSuccess extends BaseMixin(LitElement) {
...
+	_donePressed() {
+		this.changePage(PAGES.WELCOME_PAGE);
+	}
+
+	_anotherCoursePressed() {
+		this.changePage(PAGES.INPUT_PAGE);
+	}
+
+	_getCourseEnrollHref(orgUnitId) {
+		return '/d2l/lms/classlist/classlist.d2l?ou='.concat(orgUnitId);
```
Comment (josephsaffron) [inline src/components/widget/d2l-teacher-course-creation-success.js:79]: Can we use string interpolation?
```suggestion
		return `/d2l/lms/classlist/classlist.d2l?ou=${orgUnitId}`;
```

File: src/components/widget/d2l-teacher-course-creation-success.js (hunk 3 above)
Comment (TomMichielsD2L) [reply src/components/widget/d2l-teacher-course-creation-success.js:45]: 2px it is!

File: src/components/widget/d2l-teacher-course-creation-success.js [hunk 8]
```diff
@@ -28,13 +61,63 @@ class TeacherCourseCreationSuccess extends BaseMixin(LitElement) {
 
 	connectedCallback() {
 		super.connectedCallback();
+
+		if (this.pageData && 'generatedOrgUnitId' in this.pageData) {
```
Comment (TomMichielsD2L) [inline src/components/widget/d2l-teacher-course-creation-success.js:65]:
```suggestion
		if (this.pageData && this.pageData .generatedOrgUnitId) {
```

File: src/components/widget/d2l-teacher-course-creation-success.js [hunk 9]
```diff
@@ -28,13 +61,64 @@ class TeacherCourseCreationSuccess extends BaseMixin(LitElement) {
...
+	_anotherCoursePressed() {
+		this.changePage(PAGES.INPUT_PAGE);
+	}
+
+	_getCourseEnrollHref(orgUnitId) {
+		return `/d2l/lms/classlist/classlist.d2l?ou=${orgUnitId}`;
+	}
+
+	_getCourseHomepageHref(orgUnitId) {
+		return `/${orgUnitId}`;
```
Comment (josephsaffron) [inline src/components/widget/d2l-teacher-course-creation-success.js:83]:
```suggestion
		return `/d2l/home/${orgUnitId}`;
```

File: test/d2l-custom-teacher-course-creation-success.test.js [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+		}
+
+		afterEach(() => {
+			document.removeEventListener('change-page', changePageToWelcome);
+		});
+
+		it('button triggers change-page event', async() => {
+			const el = await fixture(html`<d2l-tcc-success></d2l-tcc-success>`);
+
+			document.addEventListener('change-page', changePageToWelcome);
```
Comment (TomMichielsD2L) [inline test/d2l-custom-teacher-course-creation-success.test.js:33]: Adding these to the document causes bleeding into other tests.

File: test/d2l-custom-teacher-course-creation-success.test.js [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+		function changePageToWelcome(event) {
+			expect(event.detail.page).to.equal(PAGES.WELCOME_PAGE);
+		}
+
+		afterEach(() => {
+			document.removeEventListener('change-page', changePageToWelcome);
+		});
+
+		it('button triggers change-page event', async() => {
+			const el = await fixture(html`<d2l-tcc-success></d2l-tcc-success>`);
```
Comment (TomMichielsD2L) [inline test/d2l-custom-teacher-course-creation-success.test.js:31]: Do we need to call "fixtureCleanup();" in the afterEach too?

File: test/d2l-custom-teacher-course-creation-success.test.js [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+
+			el.addEventListener('change-page', changePageToWelcome);
+
+			el.shadowRoot.querySelector('d2l-button').click();
+		});
+	});
+
+	describe('link actions', () => {
+		function changePageToInput(event) {
+			expect(event.detail.page).to.equal(PAGES.INPUT_PAGE);
```
Comment (quinnbudan) [inline test/d2l-custom-teacher-course-creation-success.test.js:41]: Is this `expect` being used?

Comment (quinnbudan) [reply test/d2l-custom-teacher-course-creation-success.test.js:41]: Never mind I meant to remove this, I see it being used below

**Pull request merged**